File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/DigitsOnly.java
package com.freight.norm;

/** Normalizer: digits only. */
public final class DigitsOnly implements Normalizer {

    @Override
    public String name() {
        return "digits_only";
    }

    @Override
    public String apply(String text) {
        StringBuilder out = new StringBuilder(text.length());
        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            if (c >= '0' && c <= '9') {
                out.append(c);
            }
        }
        return out.toString();
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/norm/digits_only.cpp
#include "freight/normalize.h"

#include <algorithm>

namespace freight {

// Normalizer: digits only.
std::string norm_digits_only(const std::string& text) {
  std::string out;
  for (size_t i = 0; i < text.size(); ++i) {
    if (text[i] >= '0' && text[i] <= '9') {
      out.push_back(text[i]);
    }
  }
  return out;
}

}  // namespace freight
